### PR TITLE
Vehicle Animation 1.0

### PR DESCRIPTION
**Pull request by srothst1**
The goal of this pull request is to add vehicle animations to the street segments in 3DStreet.

Character animations can be added to the `drive-lane` segment by adding the variant string `animated`. For instance, we have:
```
  {
    "width": 9,
    "variantString": "outbound|car|animated",
    "type": "drive-lane"
  }
```

Here are some relevant resources that I have used:
- https://github.com/3DStreet/3dstreet/pull/181
- https://docs.google.com/spreadsheets/d/1VGRxCEPN6w99BxHQoIeWVnZFDhAL4FB_JfhMOvQihxI/edit?usp=sharing

### CODE REVIEW

**Comment by srothst1** [comment]
Screen Shot 2022-06-17 at 3 16 10 PM

Here is a video showcasing the vehicle animations: https://youtu.be/kUgoRIKC6nA

**Comment by srothst1** [comment]
Hi @kfarr 👋 Please take a look at this when you get the chance.

File: src/aframe-streetmix-parsers.js [hunk 1]
```diff
@@ -295,7 +295,8 @@ function createBusElement (isOutbound, positionX) {
 }
 
 function createDriveLaneElement (variantList, positionX, segmentWidthInMeters, length, animated = 'false') {
-  var totalStreetDuration = (length / 5) * 1000;
+  var speed = 10; // meters per second
```
Comment (kfarr) [inline src/aframe-streetmix-parsers.js:298]: this increases the default speed to 10 m/s ~= 22mph

**Comment by kfarr** [comment]
Also for future reference here is reference to calculate rotations of wheel relative given a velocity and wheel radius https://www.texasgateway.org/resource/61-angle-rotation-and-angular-velocity#64411

File: src/aframe-streetmix-parsers.js [hunk 2]
```diff
@@ -294,7 +294,17 @@ function createBusElement (isOutbound, positionX) {
   return busParentEl;
 }
 
-function createDriveLaneElement (variantList, positionX, segmentWidthInMeters, length) {
+function createDriveLaneElement (variantList, positionX, segmentWidthInMeters, length, animated = 'false') {
```
[the remark this answers is not in the document]
Comment (srothst1) [reply src/aframe-streetmix-parsers.js]: No good reason - not using a boolean actually makes the code less clean. Just added a commit that changes `animated` to a boolean value in `createSidewalkClonedVariants` and `createDriveLaneElement` 👍

File: src/aframe-streetmix-parsers.js (hunk 1 above)
Comment (srothst1) [reply src/aframe-streetmix-parsers.js:298]: very speedy 🐎

**Comment by srothst1** [comment]
Here is another resource that might be helpful: https://www.omnicalculator.com/everyday-life/rpm

A 2014 Toyota Camry has tires with a [diameter of 32 inches](https://www.goodyear.com/en-US/tires/toyota/camry/2014). Thus, if we are traveling at 22mph, our rpm should be roughly 230 according to omnicalculator.

Screen Shot 2022-06-24 at 10 15 25 AM

This seems a little bit high to me, but I would be curious to see how the animation looks.

**Comment by srothst1** [comment]
There should be one full wheel rotation every `02.60869` seconds. This rate is that of a car that is navigating at 1 m/s.

**Pull request merged**